### PR TITLE
Activity for displaying all available games

### LINKED ISSUE #5: List of the available games
Global list of games
You can click on a game and see his description
Under the description there is a button "Register"

- [ ] Requires game database

### PR DESCRIPTION
**Pull request by rykeuse**
Closes #5 

- Show list of available games
- Game cards react to click and start GameViewerActivity

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        return games.size();
+    }
+
+    public static class GameHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private TextView title, location, universe;
+        private RecyclerViewClickListener mListener;
+
+        public GameHolder(View itemView, RecyclerViewClickListener listener) {
+            super(itemView);
+            title = itemView.findViewById(R.id.gameTitle);
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: Can you please use butterknife instead?

File: app/src/main/java/ch/epfl/sweng/erpa/model/Game.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class Game {
+    private String title;
+    private String location;
+    private String universe;
+    private UUID uuid = UUID.randomUUID();
+
+    public Game() {
+        title = "test";
+        location = "Lausanne";
+        universe = "DnD";
```
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/model/Game.java]: It's a dummy class that's replacing @DonnA98 until it's merged, it's not gonna be there when I close my PR

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import ch.epfl.sweng.erpa.R;
+import ch.epfl.sweng.erpa.listeners.RecyclerViewClickListener;
+
+public class GameAdapter extends RecyclerView.Adapter<GameAdapter.GameHolder> {
+    private ArrayList<Game> games;
+    private RecyclerViewClickListener mListener;
+
+    public GameAdapter(ArrayList<Game> games, RecyclerViewClickListener listener) {
+        this.games = games;
```
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: Not an ArrayList

File: app/build.gradle [hunk 4]
```diff
@@ -61,8 +61,13 @@ dependencies {
     androidTestImplementation 'org.mockito:mockito-android:2.22.0'
     annotationProcessor 'com.github.stephanenicolas.toothpick:toothpick-compiler:1.1.3'
     annotationProcessor 'com.jakewharton:butterknife-compiler:8.8.1'
-    implementation 'com.android.support:appcompat-v7:28.0.0'
+    implementation "com.kosprov.jargon2:jargon2-api:1.1.1"
+    implementation "com.kosprov.jargon2:jargon2-native-ri-backend:1.1.1"
+    implementation 'at.favre.lib:bcrypt:0.5.0'
```
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/build.gradle]: It is sorted

[the remark this answers is not in the document]
Comment (rykeuse) [reply app/build.gradle]: It's just that " comes before ' apparently

[the remark this answers is not in the document]
Comment (rykeuse) [reply app/build.gradle]: (Vim says so anyways)

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: K

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            itemView.setOnClickListener(this);
+        }
+
+        public void setDetails(Game game) {
+            title.setText(game.getName());
+            location.setText("Lausanne");
+            universe.setText(game.getUniverse());
+        }
+
+        @Override public void onClick(View v) {
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: CodeClimate reports this code is not executed. Maybe add a simple test (even if what's supposed to do it's not ready yet) and we just add a `TODO` of what it's supposed to do once we implement it

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java [hunk 6]
```diff
@@ -34,6 +37,8 @@
             RemoteServicesProviderCoordinator.class
     ).collect(Collectors.toSet());
 
+    List<Game> sampleListOfGames = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: I don't understand your question.

File: app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package ch.epfl.sweng.erpa.activities;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+
+import ch.epfl.sweng.erpa.R;
+
+public class GameViewerActivity extends AppCompatActivity {
```
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java]: It's an empty dummy class, should I just write a useless test for it?

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java (hunk 5 above)
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: Still need to write extra tests, hence the WIP mention

Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: Done, I think

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            location = gameLocation;
+            universe = universeName;
+            mListener = listener;
+            itemView.setOnClickListener(this);
+        }
+
+        public void setDetails(Game game) {
+            title.setText(game.getName());
+            location.setText("Lausanne");
+            universe.setText(game.getUniverse());
```
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: Not sure I understand why (or how, but maybe knowing why will help with that)

File: app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java]: Actually tested by the click test

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (roosemberth) [reply app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Why do you use a List when you can use a Set? Do you need these elements to be ordered?

File: app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (roosemberth) [reply app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java]: Nah, it's fine

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (roosemberth) [reply app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Resolved off-channel, will refactor once we implement a proper `GameProviderService`

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java [hunk 9]
```diff
@@ -26,21 +29,32 @@
...
+
+    List<Game> sampleListOfGames = new ArrayList<>();
+
     @Override
     public void onCreate() {
         super.onCreate();
 
         // Initialise Dependency Injection framework
-        Toothpick.setConfiguration(Configuration.forProduction().disableReflection().preventMultipleRootScopes());
+        Toothpick.setConfiguration(
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Can you please undo this change and fix you editor configuration?

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java [hunk 10]
```diff
@@ -26,21 +29,32 @@
...
+                Configuration.forProduction().disableReflection().preventMultipleRootScopes());
         FactoryRegistryLocator.setRootRegistry(new ch.epfl.sweng.erpa.smoothie.FactoryRegistry());
-        MemberInjectorRegistryLocator.setRootRegistry(new ch.epfl.sweng.erpa.smoothie.MemberInjectorRegistry());
+        MemberInjectorRegistryLocator.setRootRegistry(
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Please undo this change aswell

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java [hunk 11]
```diff
@@ -51,8 +65,10 @@ public void installModules(Scope appScope) {
 
         // Publish collection instances
         appScope.installModules(new Module() {{
-            bind(Set.class).withName(RES_REMOTE_SERVICES_PROVIDERS).toInstance(remoteServicesProviders);
+            bind(Set.class).withName(RES_REMOTE_SERVICES_PROVIDERS).toInstance(
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Please undo this change and fix you editor configuration
